### PR TITLE
add client configuration of trace analytics

### PR DESCRIPTION
**Pull request by rochdev**
This PR adds client configuration of Trace Analytics with the following options:

* Option on the tracer to globally enable/disable Trace Analytics.
* Option on every plugin to enable/disable Trace Analytics and set the default sample rate for the plugin or rate per operation name.
* Option on `tracer.trace()` to enable/disable Trace Analytics or set a specific sample rate.
* Tag that can be set on the span to set the sample rate.

### CODE REVIEW

File: src/instrumenter.js [hunk 1]
```diff
@@ -147,7 +147,13 @@ class Instrumenter {
   }
 
   _set (plugin, meta) {
-    meta = Object.assign({ config: {} }, meta)
+    const analytics = {}
+
+    if (typeof this._tracer._tracer._analytics === 'boolean') {
```
Comment (brettlangdon) [inline src/instrumenter.js]: We only want to enable web frameworks via the global `analytics` configuration.

File: src/format.js [hunk 2]
```diff
@@ -92,6 +98,10 @@ function extractMetrics (trace, span) {
   if (spanContext._sampling.priority !== undefined) {
     trace.metrics[SAMPLING_PRIORITY_KEY] = spanContext._sampling.priority
   }
+
+  if (analyticsSampleRate >= 0 && analyticsSampleRate <= 1) {
```
[the remark this answers is not in the document]
Comment (rochdev) [reply src/format.js]: You shall not pass.

File: src/instrumenter.js (hunk 1 above)
Comment (rochdev) [reply src/instrumenter.js]: Yes, the plugins are configured to only use the default rate of 1 when it's a web framework.

**Comment by rochdev** [comment]
It would be redundant since it would just be to copy the `true` or `false` value from the code to the test.

**Pull request merged**